core/src/io/anuke/mindustry/world/blocks/storage/CoreBlock.java

```diff
@@ -150,8 +150,9 @@ public void update(Tile tile){
 
             entity.heat = Mathf.lerpDelta(entity.heat, 1f, 0.1f);
             entity.time += entity.delta();
-            entity.progress += 1f / state.rules.respawnTime * entity.delta();
-
+            if(entity.currentUnit.respawns > 0){
```

Comment:
First of all, players can still respawn in `MechPad`s with 0 respawns. Second of all, I don't think the respawning blocks should handle this logic-- players should simply not be placed in respawn points when their respawns run out. This can be done by not calling `Player#updateRespawning` when the player has <= 0 respawns.